### DISCUSSION TITLE
Get the list of classes parsing the files declared on `content` inside `module.exports`

### DISCUSSION THREAD

**Discussion by Paul-Bob** [tailwindlabs/tailwindcss#11974] [Help]
Greetings! I am presently engaged in the process of integrating Tailwind CSS between a ruby gem (Avo) and its host RoR app. One of the methods that we are contemplating involves obtaining a comprehensive list of all the Tailwind CSS classes that have been declared on the paths inserted inside `module.exports` -> `content`.

From [here](https://tailwindcss.com/docs/content-configuration#configuring-source-paths)  

> Tailwind CSS works by scanning all of your HTML, JavaScript components, and any other template files for class names, then generating all of the corresponding CSS for those styles.

Now, the question arises: Is possible to invoke this "scanner" and get the list of the fetched class names?

Thank you!

**Comment by adrianthedev** [comment]
To give you a bit of context, in [our package](https://github.com/avo-hq/avo) we want to build a plugin system where anyone can write Tailwind CSS and have it loaded in the main app. The issue with that is that specificity problems pop up between each Tailwind-built CSS file.

Our take on this is to run this "scan" command in each plugin pull out a `whitelist.txt` file with the classes used by Tailwind and in the main app, the regular compilation process runs with those whitelisted classes present.
This way we won't have specificity issues between each Tailwind asset.

We're open to different approaches to building plugins based on Tailwind CSS.

Thank you!
